vendor/github.com/docker/distribution/registry/api/errcode/handler.go

```diff
@@ -36,9 +36,5 @@ func ServeJSON(w http.ResponseWriter, err error) error {
 
 	w.WriteHeader(sc)
 
-	if err := json.NewEncoder(w).Encode(err); err != nil {
-		return err
-	}
-
-	return nil
+	return json.NewEncoder(w).Encode(err)
```

Comment:
this is _really_ close to what we had, nice @dims 🙃

Reply:
LOL